web/js/link/wv.link.ui.js

```diff
@@ -106,6 +154,68 @@ wv.link.ui = wv.link.ui || function(models, config) {
         $('#permalink_content').val(models.link.get());
         $dialog.dialog("open");
         setTimeout(updateLink, 500);
+        // When an icon-link is clicked, replace the URL with current encoded link.
+        $(".icon-link").on("click", function() {
+            var fullEncodedLink = encodeURIComponent(models.link.get());
+            var promise = models.link.shorten();
+
+            // Set Facebook
+            var fbLink = document.getElementById("fb-share");
+            fbLink.setAttribute("href", "https://www.facebook.com/dialog/share?" +
+                "app_id=" + fbAppId +
+                "&href=" + fullEncodedLink +
+                "&redirect_uri=" + fullEncodedLink +
+                "&display=popup"
+            );
+
+            // Set Twitter
+            var twLink = document.getElementById("tw-share");
```

Comment:
Try to declare vars at top of scope:
`var promise, fbLink, twLink;`
or 
var promise;
var fbLink;


This approach is a more literal representation of how the code is read.